com.hazelcast.jet.core.CancellationTest.when_jobCancelledFromClient_then_jobStatusIsSetEventually

_master_ (commit 55958e30a94d419f740be754876f94fc8dc58f1f)

Failed on IBM JDK8: http://jenkins.hazelcast.com/job/jet-oss-master-ibm-jdk8/248/testReport/com.hazelcast.jet.core/CancellationTest/when_jobCancelledFromClient_then_jobStatusIsSetEventually/

Stacktrace:
```
java.lang.AssertionError: Call counter should eventually start being incremented.
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.jet.core.CancellationTest.lambda$assertExecutionStarted$4(CancellationTest.java:375)
	at com.hazelcast.jet.core.CancellationTest$$Lambda$1409.000000002D6EA9A0.run(Unknown Source)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1356)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1373)
	at com.hazelcast.jet.core.CancellationTest.assertExecutionStarted(CancellationTest.java:375)
	at com.hazelcast.jet.core.CancellationTest.when_jobCancelledFromClient_then_jobStatusIsSetEventually(CancellationTest.java:166)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
	at java.lang.reflect.Method.invoke(Method.java:508)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:818)
```

Standard output:
```
Hiccups measured while running test 'when_jobCancelledFromClient_then_jobStatusIsSetEventually(com.hazelcast.jet.core.CancellationTest):'
01:57:05, accumulated pauses: 449 ms, max pause: 22 ms, pauses over 1000 ms: 0
01:57:10, accumulated pauses: 682 ms, max pause: 311 ms, pauses over 1000 ms: 0
01:57:15, accumulated pauses: 707 ms, max pause: 67 ms, pauses over 1000 ms: 0
01:57:20, accumulated pauses: 289 ms, max pause: 21 ms, pauses over 1000 ms: 0
```

There is another test (`when_jobCancelled_then_jobStatusIsSetDuringCancellation`) in this test class which seems to have similar issue on Zulu JDK 8 and can have the same root cause: http://jenkins.hazelcast.com/job/jet-oss-master-zulu-jdk8/256/testReport/junit/com.hazelcast.jet.core/CancellationTest/when_jobCancelled_then_jobStatusIsSetDuringCancellation/

Stacktrace:
```
java.lang.AssertionError: Call counter should eventually start being incremented.
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.jet.core.CancellationTest.lambda$assertExecutionStarted$4(CancellationTest.java:375)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1356)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1373)
	at com.hazelcast.jet.core.CancellationTest.assertExecutionStarted(CancellationTest.java:375)
	at com.hazelcast.jet.core.CancellationTest.when_jobCancelled_then_jobStatusIsSetDuringCancellation(CancellationTest.java:209)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Hiccups measured while running test 'when_jobCancelled_then_jobStatusIsSetDuringCancellation(com.hazelcast.jet.core.CancellationTest):'
01:53:40, accumulated pauses: 994 ms, max pause: 416 ms, pauses over 1000 ms: 0
01:53:45, accumulated pauses: 2821 ms, max pause: 862 ms, pauses over 1000 ms: 0
01:53:50, accumulated pauses: 1626 ms, max pause: 301 ms, pauses over 1000 ms: 0
```

Another similar failure, this time in `when_jobCancelledFromClient_then_terminatedEventually` on Oracle JDK 8: http://jenkins.hazelcast.com/job/jet-oss-master-oracle-jdk8/264/testReport/junit/com.hazelcast.jet.core/CancellationTest/when_jobCancelledFromClient_then_terminatedEventually/

Stacktrace:
```
java.lang.AssertionError: Call counter should eventually start being incremented.
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.jet.core.CancellationTest.lambda$assertExecutionStarted$4(CancellationTest.java:375)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1356)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1373)
	at com.hazelcast.jet.core.CancellationTest.assertExecutionStarted(CancellationTest.java:375)
	at com.hazelcast.jet.core.CancellationTest.when_jobCancelledFromClient_then_terminatedEventually(CancellationTest.java:144)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Hiccups measured while running test 'when_jobCancelledFromClient_then_terminatedEventually(com.hazelcast.jet.core.CancellationTest):'
13:27:25, accumulated pauses: 2229 ms, max pause: 835 ms, pauses over 1000 ms: 0
13:27:30, accumulated pauses: 1936 ms, max pause: 254 ms, pauses over 1000 ms: 0
13:27:35, accumulated pauses: 3385 ms, max pause: 1299 ms, pauses over 1000 ms: 1
13:27:40, accumulated pauses: 960 ms, max pause: 496 ms, pauses over 1000 ms: 0
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/CancellationTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/CancellationTest.java
@@ -54,6 +54,8 @@ import static org.junit.Assert.assertTrue;
 @RunWith(HazelcastSerialClassRunner.class)
 public class CancellationTest extends JetTestSupport {
 
+    private static final int ASSERTION_TIMEOUT_SECONDS = 15;
+
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
 
@@ -127,7 +129,7 @@ public class CancellationTest extends JetTestSupport {
         job.cancel();
 
         // Then
-        assertJobStatusEventually(job, JobStatus.FAILED, 3);
+        assertJobStatusEventually(job, JobStatus.FAILED, ASSERTION_TIMEOUT_SECONDS);
     }
 
     @Test
@@ -169,7 +171,7 @@ public class CancellationTest extends JetTestSupport {
         job.cancel();
 
         // Then
-        assertJobStatusEventually(job, JobStatus.FAILED, 3);
+        assertJobStatusEventually(job, JobStatus.FAILED, ASSERTION_TIMEOUT_SECONDS);
     }
 
     @Test
@@ -212,11 +214,11 @@ public class CancellationTest extends JetTestSupport {
         job.cancel();
 
         // Then
-        assertJobStatusEventually(job, JobStatus.COMPLETING, 3);
+        assertJobStatusEventually(job, JobStatus.COMPLETING, ASSERTION_TIMEOUT_SECONDS);
 
         resetPacketFiltersFrom(instance1.getHazelcastInstance());
 
-        assertJobStatusEventually(job, JobStatus.FAILED, 3);
+        assertJobStatusEventually(job, JobStatus.FAILED, ASSERTION_TIMEOUT_SECONDS);
     }
 
     @Test
@@ -289,7 +291,7 @@ public class CancellationTest extends JetTestSupport {
         DAG dag = new DAG();
         dag.newVertex("blocking", BlockingProcessor::new).localParallelism(1);
         jet.newJob(dag);
-        assertTrueEventually(() -> assertTrue(BlockingProcessor.hasStarted), 3);
+        assertTrueEventually(() -> assertTrue(BlockingProcessor.hasStarted), ASSERTION_TIMEOUT_SECONDS);
         if (graceful) {
             jet.shutdown();
         } else {
@@ -304,7 +306,7 @@ public class CancellationTest extends JetTestSupport {
         DAG dag = new DAG();
         dag.newVertex("blocking", BlockingProcessor::new).localParallelism(1);
         Job job = jet.newJob(dag);
-        assertTrueEventually(() -> assertTrue(BlockingProcessor.hasStarted), 3);
+        assertTrueEventually(() -> assertTrue(BlockingProcessor.hasStarted), ASSERTION_TIMEOUT_SECONDS);
         job.cancel();
         assertBlockingProcessorEventuallyNotRunning();
     }
@@ -373,7 +375,7 @@ public class CancellationTest extends JetTestSupport {
     private static void assertExecutionStarted() {
         final long first = StuckSource.callCounter.get();
         assertTrueEventually(() -> assertTrue("Call counter should eventually start being incremented.",
-                first != StuckSource.callCounter.get()), 3);
+                first != StuckSource.callCounter.get()), ASSERTION_TIMEOUT_SECONDS);
     }
 
     private static void assertExecutionTerminated() {
@@ -384,14 +386,14 @@ public class CancellationTest extends JetTestSupport {
             previous[0] = current;
             assertTrue("Call counter should eventually stop being incremented.", current == last);
             sleepMillis(200);
-        }, 3);
+        }, ASSERTION_TIMEOUT_SECONDS);
     }
 
     private static void assertBlockingProcessorEventuallyNotRunning() {
         assertTrueEventually(() -> assertTrue(
                 String.format("BlockingProcessor should be started and done; hasStarted=%b, isDone=%b",
                         BlockingProcessor.hasStarted, BlockingProcessor.isDone),
-                BlockingProcessor.hasStarted && BlockingProcessor.isDone), 3);
+                BlockingProcessor.hasStarted && BlockingProcessor.isDone), ASSERTION_TIMEOUT_SECONDS);
     }
 
     private static class StuckSource extends AbstractProcessor {

